src/cockpit/channels/dbus.py

```diff
@@ -346,10 +350,9 @@ async def do_call(self, message):
             # If the method call has kicked off any signals related to
             # watch processing, wait for that to be done.
             async with self.watch_processing_lock:
-                # TODO: stop hard-coding the endian flag here.
                 self.send_json(
                     reply=[reply.get_body()], id=cookie,
-                    flags="<" if flags is not None else None,
+                    flags=self.endianness if flags is not None else None,
```

Comment:
I have some trouble understanding what `flags` does here. I don't see it explained in the [sd_bus_message_new_method_call doc](https://www.freedesktop.org/software/systemd/man/latest/sd_bus_message_new_method_call.html#).

But still this looks upside down: I'm fairly sure that we don't want to entirely ignore `flags` here, i.e. no main it should have said
```
flags="<" if flags is None else flags,
```

and consequently, in this PR:
```
flags=self.endianess if flags is None else flags
```

@allisonkarlitskaya can you please double-check my logic here? Thanks!

Reply:
My understanding was that the `flags` is a cockpit specific thing: https://github.com/cockpit-project/cockpit/blob/main/doc/protocol.md?plain=1#L468-L474 

It seems to implemented in the python bridge in the similar way as it's implemented in the C bridge: https://github.com/cockpit-project/cockpit/blob/main/src/bridge/cockpitdbusjson.c#L1122-L1131

Reply:
Ah right, thanks. But this pretty much validates my claim that both the old and the new condition here is the wrong way around. So this definitively needs fixing.